src/revisions_parser.py

```diff
@@ -2,37 +2,120 @@
+            # swap year and day
+            tmp = splitted[0]
+            splitted[0] = splitted[2]
+            splitted[2] = tmp
 
-    # find end
+            # use number for months
+            splitted[1] = month_to_number(splitted[1])
+    elif "." in input:
+        splitted = input.split(".")
+        if len(splitted) != 3:
+            return input
+
+        # swap year and day
+        tmp = splitted[0]
+        splitted[0] = splitted[2]
+        splitted[2] = tmp
+    else:
+        return input
+
+    return "-".join(splitted)
+
+
+def parse_general(input: str, regex: str, ver_index: int, date_index: int) -> List[Dict[str, str]]:
     end = re.search(r"\n{4}", input, re.IGNORECASE)
     if end:
         input = input[:end.span(0)[1]]
     else:
         input = input[:5000]
-    
-    result = re.findall(r"\s+([0-9.]+)\s+([0-9-]+)?\s+(.*)", input)
-    
+
+    result = re.findall(regex, input)
+
+    final_result = []
     for i in range(len(result)):
-        result[i] = {
-            "version": result[i][0],
-            "date": result[i][1],
-            # there can be more information, like author...
-            "description": result[i][2].split("  ")[-1],
-        }
+        version = result[i][ver_index]
+        date = clean_date(result[i][date_index])
+        # there can be more information, like author...
+        description = result[i][2].split("  ")[-1]
+
+        final_result.append({
+            "version": version,
+            "date": date,
+            "description": description,
+        })
+    return final_result
+
+
+def parse_rev_date_desc(input: str) -> List[Dict[str, str]]:
+    return parse_general(input, r"\s+v?([0-9.]+)\s+([0-9-A-Za-z-\.]+)?[\s:]\s+(.*)", 0, 1)
+
+
+def parse_date_ver_desc(input: str) -> List[Dict[str, str]]:
+    return parse_general(input, r"\s+v?([0-9-A-Za-z-\.]+)?\s+([0-9.]+)[\s:]\s+(.*)", 1, 0)
+
+
+def parse(input: str) -> List[Dict[str, str]]:
+    found = re.search(r"^\w*rev\w*\s+date\s+.*description",
+                      input, re.IGNORECASE | re.MULTILINE)
+    if found:
+        input = input[found.span(0)[1]:]
+
+        return parse_rev_date_desc(input)
+
+    found = re.search(r"date\s+ver\w*\s+.*description", input, re.IGNORECASE)
+    if found:
+        input = input[found.span(0)[1]:]
+
+        return parse_date_ver_desc(input)
+
+    found = re.search(r"version\s\s+description", input, re.IGNORECASE)
+    if found:
+        input = input[found.span(0)[1]:]
+
+        return parse_rev_date_desc(input)
+
+    # final attempt, filtering out capitalization that could be used in normal text
+    iter = re.finditer(r"REVISION HISTORY|Revision [Hh]istory|VERSION CONTROL|Version [Cc]ontrol", input)
+    start = None
+    a = 0
+    for start in iter:
+        a += 1
+        # first mention probably in toc, so take the second one, if there is one
+        if a == 2:
+            break
+    if start:
+        input = input[start.span(0)[1]:]
```

Comment:
Would it not be better to write this more functionally? What you want is at most first two elements of `iter`, then take the last one of this subsequence. The first 7 lines should then reduce to just one.